### ISSUE TITLE
Invalid unit_personality's current_focus and undistracted_focus on 32 bits platforms

### ISSUE THREAD

**Issue by cvuchener** [DFHack/df-structures#299]
Expected values are 98 and 96 (4*24 needs).
```
<unit_personality: 0xcf815b20>
values                 	 = <vector<unit_personality.T_values*>[4]: 0xcf815b20>
unk_v40_2              	 = <vector<unit_personality.T_unk_v40_2*>[0]: 0xcf815b2c>
emotions               	 = <vector<unit_personality.T_emotions*>[2]: 0xcf815b38>
dreams                 	 = <vector<unit_personality.T_dreams*>[1]: 0xcf815b44>
next_dream_id          	 = 1
unk_v40_6              	 = <vector<unit_personality.T_unk_v40_6*>[0]: 0xcf815b54>
traits                 	 = <uint16_t[]: 0xcf815b60>
civ_id                 	 = 9
cultural_identity      	 = -1
unk5                   	 = <vector<unit_personality.T_unk5*>[1]: 0xcf815bcc>
unk6                   	 = <vector<int16_t>[0]: 0xcf815bd8>
stress_level           	 = 40
unk_v4014_2            	 = 0
unk_v4014_3            	 = 2800
unk_v4019_1            	 = 2
unk_v4019_2            	 = 0
needs                  	 = <vector<unit_personality.T_needs*>[24]: 0xcf815bf8>
flags                  	 = <unit_personality.T_flags: 0xcf815c04>
temporary_trait_changes	 = nil
unk_v4201_3            	 = -1
unk_v4201_4            	 = -1
unk_v4410_1            	 = 0
unk_v4410_2            	 = 98
current_focus          	 = 96
undistracted_focus     	 = -828746448
```

The change I made in #264 was not the good one. Could unk_v4410_1 and unk_v4410_2 be a single pointer? 

-828746448 must be the correct value for the performance_skills pointer. When I try to read the current field, it is null or invalid:
```
[DFHack]# :lua ~dfhack.gui.getSelectedUnit().status.current_soul.performance_skills
nil
```
another unit:
```
[DFHack]# :lua ~dfhack.gui.getSelectedUnit().status.current_soul.performance_skills
<unit_soul.T_performance_skills: 0xaf0200>
nil
stack traceback:
	[C]: in ?
	[C]: in function 'tostring'
	./hack/lua/dfhack.lua:169: in function 'printall'
	...e/clement/games/df/df_44_12_linux32/hack/scripts/lua.lua:68: in local 'script_code'
	./hack/lua/dfhack.lua:680: in function 'dfhack.run_script_with_env'
	(...tail calls...)
```
(DFHack command line is broken after that)

These tests were done on linux 32 bits, but I have the same issues with the Dwarf Therapist memory layouts I generated for windows 32 bits.

**Comment by PatrikLundell** [comment]
On my 64 bit version (on Windows 10.1, DF 0.44.12) I find lots of units that have a performance_skills pointer with a nil value. Most of them are animals or invader monsters (trolls, etc), but a number of them have two part names (and I don't think I have that many pets). I believe I've seen the structure being absent on residents earlier (although the memory is not to be relied on). Thus, I suspect this pointer (or, rather, what it should point to) only gets allocated once there is any skill to report.

I agree the current_focus/undistracted_focus pair look like they're shifted in your case, though.

**Comment by quietust** [comment]
This appears to have been fixed - unk_v4410_1 was changed from an int32 to a pointer and unk_v4410_2 was removed (which fixed alignment on 32-bit without breaking it on 64-bit).

**Comment by lethosor** [comment]
70e75246c04878e77ab037711cc2b0d5cbd62819, for reference

**Comment by cvuchener** [comment]
I used a similar change for generating DT memory layouts (and forgot about it) and no issue were reported since (although I am not sure the 32 bits version is much tested). I regenerated them using the latest master instead of my old fix and the only change is the disappearance of OnBreak enum value from the same commit. So yes this is fixed by 70e7524.
